docs/docsite/rst/dev_guide/developing_module_utilities.rst

```diff
@@ -4,84 +4,112 @@
 Appendix: Module Utilities
 **************************
 
-Ansible provides a number of module utilities that provide helper functions that you can use when developing your own modules. The ``basic.py`` module utility provides the main entry point for accessing the Ansible library, and all Python Ansible modules must, at minimum, import ``AnsibleModule``::
+Ansible provides a number of module utilities, or snippets of shared code, that
+provide helper functions you can use when developing your own modules. The
+``basic.py`` module utility provides the main entry point for accessing the
+Ansible library, and all Python Ansible modules must, at minimum, import
+``AnsibleModule``::
 
   from ansible.module_utils.basic import AnsibleModule
 
+The ``ansible.module_utils`` namespace is not a plain Python package: it is
+constructed dynamically for each task invocation, by extracting imports and
+resolving those matching the namespace against a search path derived from the
+active configuration.
+
 If you need to share Python code between some of your own local modules, you can use Ansible's ``module_utils`` directories for this. When you run ``ansible-playbook``, Ansible will merge any files in the local ``module_utils`` directory into the ``ansible.module_utils`` namespace. For example, if you have your own custom modules that import a ``my_shared_code`` library, you can place that into a ``./module_utils/my_shared_code.py`` file in the root location where your playbook lives, and then import it in your modules like so::
 
   from ansible.module_utils.my_shared_code import MySharedCodeClient
 
 Your custom ``module_utils`` directories can live in the root directory of your playbook, or in the individual role directories, or in the directories specified by the ``ANSIBLE_MODULE_UTILS`` configuration setting.
 
-Ansible ships with the following list of ``module_utils`` files. The module utility source code lives in the ``./lib/ansible/module_utils`` directory under your main Ansible path - for more details on any specific module utility, please see the source code.
+Naming and finding module utilities
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Ansible uses namespacing to organize module utilities. We store generic
+utilities in the root directory. For example,
```

Comment:
We're starting to push generic utilities into the ansible/module_utils/common/ subdirectory.  See especially sdoran's work around splitting apart basic.py.  I believe we'll be moving the other generic modules like urls.py into common as well (_text.py has already been moved in 2.8).

Reply:
+1 to the new wording